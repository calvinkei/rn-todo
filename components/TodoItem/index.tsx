import React from 'react';
import {Button, StyleSheet, Text, View} from 'react-native';

interface TodoItemProps {
  title: string;
  onRemove: () => void;
  onPress: () => void;
}

function TodoItem({title, onRemove, onPress}: TodoItemProps): JSX.Element {
  return (
    <View style={styles.container}>
      <Text testID="title" onPress={onPress} style={styles.input}>
        {title}
      </Text>
      <Button testID="remove button" title="Remove" onPress={onRemove} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#fff',
    paddingHorizontal: 12,
    paddingVertical: 8,
    margin: 12,
    borderRadius: 12,
  },
  input: {
    flex: 1,
    backgroundColor: '#fff',
    fontSize: 16,
    fontWeight: '600',
  },
});

export default TodoItem;
