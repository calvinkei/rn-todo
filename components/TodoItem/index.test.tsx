import 'react-native';
import React from 'react';
import TodoItem from '.';
import {it, jest, expect} from '@jest/globals';
import renderer from 'react-test-renderer';

it('matches previous snapshot', () => {
  const onRemove = jest.fn();
  const onPress = jest.fn();
  const tree = renderer
    .create(<TodoItem onRemove={onRemove} onPress={onPress} title="title" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('calls onPress when title is clicked', () => {
  const onRemove = jest.fn();
  const onPress = jest.fn();
  const instance = renderer.create(
    <TodoItem onRemove={onRemove} onPress={onPress} title="title" />,
  );
  instance.root.findByProps({testID: 'title'}).props.onPress();
  expect(onPress).toBeCalledTimes(1);
});

it('calls onRemove when remove button is clicked', () => {
  const onRemove = jest.fn();
  const onPress = jest.fn();
  const instance = renderer.create(
    <TodoItem onRemove={onRemove} onPress={onPress} title="title" />,
  );
  instance.root.findByProps({testID: 'remove button'}).props.onPress();
  expect(onRemove).toBeCalledTimes(1);
});
