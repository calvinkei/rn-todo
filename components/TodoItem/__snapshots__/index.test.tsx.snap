// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches previous snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 12,
      "flexDirection": "row",
      "margin": 12,
      "paddingHorizontal": 12,
      "paddingVertical": 8,
    }
  }
>
  <Text
    onPress={[MockFunction]}
    style={
      {
        "backgroundColor": "#fff",
        "flex": 1,
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
    testID="title"
  >
    title
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="remove button"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Remove
      </Text>
    </View>
  </View>
</View>
`;
